frame/staking/src/migrations.rs

```diff
@@ -31,9 +31,24 @@ pub mod v10 {
 	impl<T: Config> OnRuntimeUpgrade for MigrateToV10<T> {
 		fn on_runtime_upgrade() -> frame_support::weights::Weight {
 			if StorageVersion::<T>::get() == Releases::V9_0_0 {
+				// apply any pending slashes that where queued -- means we might slash someone a bit
+				// too early, but we will definitely won't forget to slash them. The cap of 512 is
+				// somewhat randomly taken to prevent us from iterating over an arbitrary large
+				// number of keys `on_runtime_upgrade`.
```

Comment:
This should be documented in the docs of the `MigrateToV10` struct. So people can see this.